scripts/formater.pl

```diff
@@ -32,21 +32,29 @@
 #
 # NP: The Squid code requires astyle version 2.04 (exactly for now)
 #
-my $ASTYLE_BIN="/usr/local/bin/astyle";
-if (! -x $ASTYLE_BIN) {
-  $ASTYLE_BIN="/usr/bin/astyle";
-}
-if (! -x $ASTYLE_BIN) {
-  $ASTYLE_BIN="/usr/local/src/astyle-2.04/bin/astyle";
+my @ASTYLE_PATHS=(
+    $ENV{'ASTYLE'},
+    "/usr/local/bin/astyle",
+    "/usr/bin/astylee",
+    "/usr/local/src/astyle-2.04/bin/astyle"
+);
```

Comment:
We do not need an ever-growing list of hard-coded astyle locations anymore. Those who want to specify an exact location should use the new command line option. Others will use `$PATH`.

Reply:
My undertanding is that PERL doesn't support $PATH, otherwise why go through the effort?

Reply:
Perl supports $PATH. We "go through the effort" because Amos insisted on hard-coding the location of the "right" astyle executable. It makes some of his packaging tasks easier. I sincerely hope that hack is no longer needed because the location can now be specified on the command line, overwriting any $PATH searches.

Reply:
... and the limitations around juggling west+east+eu OS scripting are no longer relevant. So this path checking can be removed any time. All that matters nowdays is the version check passing when the executable is being auto-detected.